pkg/daemon/server/service/pipeline_metrics_query.go

```diff
@@ -238,22 +207,23 @@ func (ps *pipelineMetadataQuery) GetVertexMetrics(ctx context.Context, req *daem
 					}
 				}
 			}
+			vm := &daemon.VertexMetrics{
+				Pipeline: &ps.pipeline.Name,
+				Vertex:   req.Vertex,
+				Pendings: pendings,
+			}
 
-			if !ps.useNewRateCalculation {
-				metricsArr[i] = &daemon.VertexMetrics{
-					Pipeline:        &ps.pipeline.Name,
-					Vertex:          req.Vertex,
-					ProcessingRates: processingRates,
-					Pendings:        pendings,
-				}
+			// Get the processing rate for this partition
+			if abstractVertex.IsReduceUDF() {
+				// the processing rate of this ith partition is the rate of the corresponding ith pod.
+				vm.ProcessingRates = ps.rater.GetPodRates(req.GetVertex(), int(i))
 			} else {
-				metricsArr[i] = &daemon.VertexMetrics{
-					Pipeline:        &ps.pipeline.Name,
-					Vertex:          req.Vertex,
-					ProcessingRates: vertexLevelRates,
-					Pendings:        pendings,
-				}
+				// if the vertex is not a reduce udf, then the processing rate is the sum of all pods in this vertex.
+				// TODO - change this to display the processing rate of each partition when we finish multi-partition support for non-reduce vertices.
```

Comment:
```suggestion
				// TODO (multi-partition) - change this to display the processing rate of each partition when we finish multi-partition support for non-reduce vertices.
```

we have too many TODOs, tagging with `multi-partition` might help while searching :)